File: DynProg/uniquePaths.py
class Solution:
    
    def recursePaths(self, grid: List[List], m: int, n: int) -> int:
        if(grid[m][n] != 0):return grid[m][n]
        
        if(m < len(grid) - 1):
            grid[m][n] = self.recursePaths(grid, m+1, n)
            
        if(n < len(grid[0]) -1):
            grid[m][n] += self.recursePaths(grid, m, n+1)
            
        return grid[m][n]
    
    def uniquePaths(self, m: int, n: int) -> int:
        grid = [[0 for i in range(n)] for j in range(m)]
        grid[m-1][n-1] = 1
        return self.recursePaths(grid, 0, 0)

File: DynProg/uniquePaths.java
class Solution {
    public int recurseSolution(int[][] grid, int m, int n){
        // Base Case if the star is reached
        // or If the spot has been memoized
        if(grid[m][n] != 0){return grid[m][n];}
        
        if(n < grid[0].length-1){
            grid[m][n] = recurseSolution(grid, m, n+1);
        }
        
        if(m < grid.length-1){
            grid[m][n] += recurseSolution(grid, m+1, n);
        }
        return grid[m][n];
    }
    public int uniquePaths(int m, int n) {
        int[][] grid = new int[m][n];
        grid[m-1][n-1] = 1;
        return recurseSolution(grid, 0, 0);
    }
}
